evidence/doc.go

```diff
@@ -0,0 +1,36 @@
+/*
+Package evidence handles all evidence storage and gossiping from detection to block proposal.
+For the different types of evidence refer to the `evidence.go` file in the types package.
```

Comment:
Do you think that this document should also go through all the evidence types and explain the detection and verification process as well @melekes?

Reply:
probably not

Reply:
Perhaps a link would be helpful, though.

Reply:
I will link it to the fork accountability spec at the moment, but in the future, once the evidence spec is written which explains a better overview of the entire evidence process I think we should reference that.